test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -62,18 +63,35 @@ public static void main(String[] args) throws Exception {
         try {
             // create test directory
             String tmpDir = System.getProperty("user.home");
-            testDir = new File[1];
-            testDir[0] = new File(tmpDir, "testDir");
-            if (!testDir[0].exists())
-                testDir[0].mkdir();
-            testDir[0].deleteOnExit();
-
-            // create temporary files inside testDir
-            tempFile = new File[5];
+
```

Comment:
Does this test HAVE to use the home dir to create the temporary folders and files ?
Is there some reason some part of the test absolutely requires the home directory ?
It isn't obvious to me.
Why can't you instead use  System.getProperty("java.io.tmpdir");

Reply:
There is no need to have the `home dir` to create the temporary folders and files. Since I was testing in my local mahine to create temp folders in home directory, I kept it as it is.

Changed `home dir` to `java.io.tmpdir` for creating temporary folders and files.